src/modules/avformat/producer_avformat.c

```diff
@@ -3814,33 +3816,31 @@ static int producer_get_frame(mlt_producer producer, mlt_frame_ptr frame, int in
 
 static int producer_probe(mlt_producer producer)
 {
-    producer_avformat self = producer->child;
     int error = 0;
+    mlt_properties properties = MLT_PRODUCER_PROPERTIES(producer);
 
-    // Update the video properties if the index changed
-    int video_index = pick_video_stream(self);
-
-    if (self->video_format && video_index > -1 && video_index != self->video_index)
-        self->probe_complete = 0;
-
-    // Update the audio properties if the index changed
-    int audio_index = pick_audio_stream(self);
-
-    if (self->audio_format && audio_index > -1 && audio_index != self->audio_index)
-        self->probe_complete = 0;
-
-    if (self->probe_complete) {
+    if (mlt_properties_get_int(properties, "_probe_complete")) {
         return error;
     }
 
+    if (!mlt_properties_exists(properties, "_probe_complete")) {
+        // If metadata was loaded from XML, no need to probe
+        int video_in_use = mlt_properties_get_int(properties, "vstream") > -1;
+        if (video_in_use && mlt_properties_exists(properties, "meta.media.progressive")) {
+            return error;
+        } else if (!video_in_use && mlt_properties_exists(properties, "meta.media.0.codec.name")) {
```

Comment:
On a Google Pixel phone (there is shared file for this in our S3 folder), there is no "meta.media.0.codec.name" property. In fact, stream 0 is elided because it is not audio or video, and avformat producer only cares about those types of streams. One might be inclined to test for a non-stream-specific meta.media property such as "meta.media.sample_aspect_num". However, that will not be present on audio-only file (`!video_in_use`). So, I think the only reliable way is to get "meta.media.nb_streams" and enumerate the streams and look for "meta.media.%d.stream.type".

Reply:
Actually, the presence of  "meta.media.nb_streams" might be good enough to indicate that the metadata properties have been loaded by XML. It is interesting to know that all the audio metadata is loaded when the file is opened. It is only video metadata that might require a frame to be pulled, or maybe even an image. I will change it to use nb_streams and do some testing.

Reply:
>  "meta.media.nb_streams" might be good enough to indicate that the metadata properties have been loaded by XML

Yes, that is true. The only exception would be a file with no usable audio or video, which is not only very rare, but I think would cause no harm here anyways.

Reply:
I did make the change to nb_streams. But I am not able to trigger that optimization with Shotcut. I notice that we do not store meta.media properties in the XML. Is that on purpose? It seems that all clips will at least be opened when an MLT project is opened (not not necessarily decoding a frame). Maybe there is an opportunity for a future optimization.

Reply:
Yes, the meta properties are omitted primarily to keep project file size smaller and make it easier to read when necessary. Now, I worry what will happen with proxy editing if they are included. They may initially be based on either the source or the proxy initially, but often converted to the proxy while editing. Then, when switching to the source, they will still represent the proxy, which might be different than source. However, they are recomputed when the producer is actually opened (in first get_frame when using avformat-novalidate). I do not know how that interplay's with chain's needs. That needs analysis and testing.

> It seems that all clips will at least be opened when an MLT project is opened

The point of the avformat-novalidate producer a long while ago was to avoid that, but that does not need meta properties provided. The non-chain producer approach only needs a few things like length and stream indices.  It seems your switch to chain for everything unwound that optimization.

Reply:
It looks like chain is mainly interested in frame rate, and proxy frame rate is the same as source. The problem that may exist when relying on this info is kind of the same that exists today: when source is variable frame rate (or something strange with many digits of precision) the proxy is a constant frame rate approximate. I will do the testing around this to make the decision for Shotcut.

Reply:
Your analysis is correct. A chain wants to know the producers "native" frame rate so that it can have the highest temoral resolution if it is doing speed tricks. Maybe there is a different (or less complex) way for the chain to know the frame rate without the needs to probe or pull frames.